test/test_cuda.py

```diff
@@ -1314,6 +1314,20 @@ def test_record_stream_on_shifted_view(self):
 
         self.assertNotEqual(try_realloc.data_ptr(), data_ptr)
 
+    def test_external_streams(self):
+        stream_a = torch.cuda.Stream()
+        ext_stream = torch.cuda.streams.ExternalStream(stream_a.cuda_stream)
+        self.assertEqual(stream_a.cuda_stream, ext_stream.cuda_stream)
+
+    @unittest.skipIf(not TEST_MULTIGPU, "detected only one GPU")
```

Comment:
cc @ptrblck: not for this PR, of course, but as a note for us it'd be nice to update test_cuda.py to use the device generic test suite, which has decorators for multiple devices and supports options like running the test suite on the first CUDA device